File: frontend/src/api/ai.ts
import request from "@/utils/request";
import type { AIRecommendRequest, AIRecommendResponse } from "@/types/ai";

export const fetchAIRecommend = (payload: AIRecommendRequest) =>
  request.post<AIRecommendResponse>("/ai/recommend", payload).then((res) => res.data);

File: src/api/ai.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from api.deps import get_db, get_current_user
from dtos.ai import AIRecommendDTO
from schema.ai import AIRecommendRequest, AIRecommendResponse
from services.ai_recommend_service import AIRecommendService


router = APIRouter(tags=["ai"])


@router.post("/ai/recommend", response_model=AIRecommendResponse)
def ai_recommend(
    payload: AIRecommendRequest,
    db: Session = Depends(get_db),
    current_user=Depends(get_current_user),
):
    service = AIRecommendService(db)
    try:
        items = service.recommend(current_user.id, AIRecommendDTO(**payload.model_dump()))
    except ValueError as exc:
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(exc))
    return {"items": items}
